plugins/woocommerce-admin/client/products/sections/product-shipping-section.tsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import { SelectControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { Link, useFormContext } from '@woocommerce/components';
+import {
+	EXPERIMENTAL_PRODUCT_SHIPPING_CLASSES_STORE_NAME,
+	Product,
+	ProductShippingClass,
+} from '@woocommerce/data';
+import interpolateComponents from '@automattic/interpolate-components';
+
+/**
+ * Internal dependencies
+ */
+import { ProductSectionLayout } from '../layout/product-section-layout';
+import { getTextControlProps } from './utils';
+import { ADMIN_URL } from '../../utils/admin-settings';
+
+const DEFAULT_SHIPPING_CLASS_OPTIONS: SelectControl.Option[] = [
+	{ value: '', label: __( 'No shipping class', 'woocommerce' ) },
+	{ value: '-1', label: __( 'Standard shipping', 'woocommerce' ) },
+];
+
+function mapShippingClassToSelectOption(
+	shippingClasses: ProductShippingClass[]
+): SelectControl.Option[] {
+	return shippingClasses.map( ( { id, name } ) => ( {
+		value: `${ id }`,
+		label: name,
+	} ) );
+}
+
+export const ProductShippingSection: React.FC = () => {
+	const { getInputProps } = useFormContext< Product >();
+
+	const shippingClassOptions = useSelect( ( select ) => {
+		const { getProductShippingClasses } = select(
+			EXPERIMENTAL_PRODUCT_SHIPPING_CLASSES_STORE_NAME
+		);
+		const shippingClasses =
+			getProductShippingClasses< ProductShippingClass[] >();
+		if ( Array.isArray( shippingClasses ) ) {
+			return [
+				...DEFAULT_SHIPPING_CLASS_OPTIONS,
+				...mapShippingClassToSelectOption( shippingClasses ),
+			];
```

Comment:
We tend to avoid creating any new arrays or objects within the `useSelect` function. As everything returned by the `useSelect` gets "cached". Meaning the `useSelect` will only trigger a React re-render when the returned value is different.
It only checks this by `===`, so any newly created objects or arrays will always show as dirty as the reference will be different.

In theory this is fine for your `array` as long as the order stays the same, as the `useSelect` checks for each value with the object returned.
In this case it would loop over each shipping class item and check if it matches the old one.

What I would recommend is just returning the `shippingClasses`: 
```
return {
  shippingClasses: getProductShippingClasses< ProductShippingClass[] >();
}
```
Then outside of the `useSelect` append the two arrays and map them.
Hope that makes sense, let me know if you have any questions?